implementation/scipamato/core/persistence-jooq/build.gradle.kts

```diff
@@ -0,0 +1,117 @@
+import org.flywaydb.gradle.task.FlywayCleanTask
+import org.flywaydb.gradle.task.FlywayInfoTask
+import org.flywaydb.gradle.task.FlywayMigrateTask
+
+plugins {
+    Lib.jooqPlugin().run { id(id) version version }
+    Lib.flywayPlugin().run { id(id) version version }
+}
+
+description = "SciPaMaTo-Core :: Persistence jOOQ Project"
+
+configurations {
+    create("flywayMigration")
+}
+
+dependencies {
+    api(project(Module.scipamatoCore("persistence-api")))
+    api(project(Module.scipamatoCommon("persistence-jooq")))
+    implementation(project(Module.scipamatoCore("entity")))
+    implementation(project(Module.scipamatoCommon("utils")))
+
+    jooqRuntime(Lib.postgres())
+    runtimeOnly(Lib.postgres())
+    api(Lib.jOOQ("jooq"))
+
+    implementation(Lib.springSecurity("core"))
+    implementation(Lib.commonsLang3())
+    implementation(Lib.commonsCollection())
+
+    testCompile(project(Module.scipamatoCommon("persistence-jooq-test")))
+    testCompile(project(Module.scipamatoCommon("test")))
+    testCompile(project(Module.scipamatoCore("entity")))
+
+    testCompile(Lib.lombok())
+    testAnnotationProcessor(Lib.lombok())
+
+    flywayMigration(Lib.postgres())
+    integrationTestRuntimeOnly(Lib.postgres())
+}
+
+val bootCoreProps = file("src/main/resources/application.properties").asProperties()
+val bootCoreItProps = file("src/intTest/resources/application.properties").asProperties()
+
+val generatedSourcesPath = "build/generated-src/jooq"
+sourceSets {
+    main {
+        java {
+            srcDir(setOf(generatedSourcesPath, "src/main/java"))
+        }
+    }
+}
+
+apply(from = "$rootDir/gradle/jooq-core.gradle")
+
+tasks {
+    val flywayMigrate by existing(FlywayMigrateTask::class) {
+        description = "Triggers database migrations for the main database"
+        url = bootCoreProps.getProperty("spring.datasource.url")
+        user = bootCoreProps.getProperty("spring.flyway.user")
+        password = bootCoreProps.getProperty("spring.flyway.password")
+        schemas = arrayOf(bootCoreProps.getProperty("db.schema"))
+        locations = arrayOf(bootCoreProps.getProperty("spring.flyway.locations"))
+        configurations = arrayOf("compile", "flywayMigration")
+    }
```

Comment:
Here I think I would try to extract the common part from all this task configurations.

Maybe something like this:
```kotlin
fun AbstractFlywayTask.configure(props: Properties) {
    url = bootCoreProps.getProperty("spring.datasource.url")
    user = bootCoreProps.getProperty("spring.flyway.user")
    password = bootCoreProps.getProperty("spring.flyway.password")
    schemas = arrayOf(bootCoreProps.getProperty("db.schema"))
    locations = arrayOf(bootCoreProps.getProperty("spring.flyway.locations"))
    configurations = arrayOf("compile", "flywayMigration")
}
```

And usage:
```kotlin
val flywayMigrate by existing(FlywayMigrateTask::class) {
  description = "Triggers database migrations for the main database"
  configure(bootCoreProps)
}
```

Reply:
Very nice, thanks!